### DISCUSSION TITLE
No more comparison ?

### DISCUSSION THREAD

**Discussion by Gluups** [WinMerge/winmerge#1081] [General]
Hello everybody,
I had a comparison going on between two files, that was supporting the control of a backup.
There were five differences, mainly the important part was saved OK, but the system was not saved, that was normal as it was locked.

But I had to restart the machine, and today, I suppose WinMerge has been updated, it just says "the files are different", period.
Is it completely lost, cannot I get the comparison ?

That is a problem as I must send the machine to be repaired.

**Comment by sdottaka** [comment]
Is the checkbox in the Message Boxes category of the Options dialog checked as shown below?

If the checkbox is checked, please uncheck it.

Also, if you are using an older version of WinMerge that does not display the Message Boxes category, try uninstalling and reinstalling it.


![image](https://user-images.githubusercontent.com/98126/145809309-3f244de3-21a4-4469-ad22-af4843ff5c53.png)

**Comment by Gluups** [comment]
You  mean there ?
![2021-12-13 (2)](https://user-images.githubusercontent.com/6567976/145812580-0402858a-79ba-446e-a716-4bb2a90dbbcd.png)

My version number is 2.16.10 x86

**Comment by sdottaka** [comment]
You seem to be using an older version.

Try reinstalling WinMerge or pressing the Reset button in the General category of the Options dialog.

![image](https://user-images.githubusercontent.com/98126/145813342-5574842e-948e-4af6-9483-cb53400b03cc.png)

**Comment by Gluups** [comment]
Oh got it.
The dialog box displays beyond the screen, so it requires an operational attention.
I mixed up with the dialog box that asks me whether I am sure if I want to open that big file, so I answered yes.
But it proposed me to display only the results, rather than the details, so I had to answer no.
And now I have the details, thanks.

I have a resolution of 1920x1020, with a system font of size 22.

According to the preferences dialog box, I wonder whether we have the same version.
What is yours ?

**Comment by sdottaka** [comment]
My first screenshot is version 2.16.16 and my second screenshot is version 2.16.8.

**Comment by Gluups** [comment]
Oh, so you use two versions simultaneously ?
Well, when that machine is gone, I shall update, it will be easier to ask questions ...

**Comment by sdottaka** [comment]
WinMerge also has zip versions that don't need to be installed, which allows you to launch multiple versions at the same time.

**Comment by Gluups** [comment]
Oh, so that you can do different comparisons at the same time ?

**Comment by Gluups** [comment]
Do you know how I can go to the end of a difference, or to the following difference ?
Each part has many lines, and it is not really necessary to read them all.

**Comment by sdottaka** [comment]
1. To go to the last difference block in the file, press Alt+End key.
2. Unfortunately, there is no shortcut key to move to the last in-line difference in a difference block.
However, pressing Shift+F4 key will move the in-line difference in the opposite direction, so pressing Shift+F4 on the next line of a difference block will move to the last in-line difference.

3. To move to the next difference block in the file, press Alt+Down.
4. To move to the next in-line difference in a difference block, press F4.

**Reply by Gitoffthelawn** [reply]
> Unfortunately, there is no shortcut key to move to the last in-line difference in a difference block.

What do you think about the idea of adding a shortcut?

**Reply by Gluups** [reply]
Well, the machine is back, but this sounds as a great idea.
I had even no idea it was possible.

**Reply by sdottaka** [reply]
At the moment, I'm not thinking of adding that feature and assigning shortcut keys.

**Comment by Gluups** [comment]
Hello,
Well,  Alt Up and Alt Down were a great help, thank you.
Not sure I can go from a difference block to the previous with Shift F4.
In the last difference block, I guess I have nothing more to update as it is the system. But I have to browse it to be sure. Is there a possibility to browse by directory ? The path is given at the beginning of the line.
So, a way to say it, is select a part at the beginning of a line, and you want to find the next line that does not begin with the selection.
That is a classical programming exercise and I can play up a little with that, but if it is integrated in WinMerge, then it would be a loss of time to do it again.

Oh, I had a few crashes, if I remember well it was with Shift F4. Perhaps you want to get a log, if there is one somewhere ?

**Comment by sdottaka** [comment]
I'm not sure I understand exactly what you want to do, but there is probably no efficient way to do it with WinMerge right now.  
WinMerge can search by regular expression ( `^(?!Directory).*$`  as follows, but it is tedious to type it every time.
![image](https://user-images.githubusercontent.com/98126/145984934-4438526c-70b4-40ca-9562-d034a7d71513.png)

If it crashes, could you please try a more recent version?
WinMerge does not output any logs.

**Reply by Gluups** [reply]
OK Thank you.
